src/metabase/integrations/ldap.clj

```diff
@@ -79,10 +79,15 @@
   :getter  (fn []
              (json/parse-string (setting/get-string :ldap-group-mappings) #(DN. (str %))))
   :setter  (fn [new-value]
-             (doseq [k (keys new-value)]
-               (when-not (DN/isValidDN (name k))
-                 (throw (IllegalArgumentException. (tru "{0} is not a valid DN." (name k))))))
-             (setting/set-json! :ldap-group-mappings new-value)))
+             (cond
+               (string? new-value)
+               (recur (json/parse-string new-value))
+
+               (map? new-value)
+               (do (doseq [k (keys new-value)]
+                     (when-not (DN/isValidDN (name k))
+                       (throw (IllegalArgumentException. (tru "{0} is not a valid DN." (name k))))))
+                   (setting/set-json! :ldap-group-mappings new-value)))))
```

Comment:
I've changed this because it was throwing an error when used with my revised version of `with-temporary-setting-values` (see below). The macro was reading the JSON string from the DB, changing the setting temporarily, then writing back the original value via the custom setter. But the custom setter would error because it's expecting a Clojure map.

Changing this setter to accept JSON strings fixes the error, but there still is a broader issue that `with-temporary-setting-values` requires custom setters to accept values of the same type & format as what is stored in the DB. Changing `with-temporary-setting-values` to bypass custom setters doesn't work, because it breaks tests that use the `humanization-strategy` setting which has side-effects in its custom setter (changing table & field names). Not sure if there's a clean way to avoid both issues.